internal/cli/branding_texts.go

```diff
@@ -90,164 +117,190 @@ func showBrandingTexts(cli *cli, inputs *brandingTextsInputs) func(cmd *cobra.Co
 
-func marshalBrandingTextBody(b map[string]interface{}) (string, error) {
-	bodyBytes, err := json.Marshal(b)
-	if err != nil {
-		return "", fmt.Errorf("Failed to serialize the custom texts to JSON: %w", err)
+		customTranslations, err := cli.api.Prompt.CustomText(inputs.Prompt, inputs.Language)
+		if err != nil {
+			return err
+		}
+
+		brandingTextTranslations := mergeBrandingTextTranslations(defaultTranslations, customTranslations)
+
+		for key, text := range brandingTextTranslations {
+			contentToEdit[key] = text
+		}
+
+		return nil
+	}); err != nil {
+		return "", fmt.Errorf(
+			"unable to load custom text for prompt %s and language %s: %w",
+			inputs.Prompt,
+			inputs.Language,
+			err,
+		)
 	}
 
-	var buf bytes.Buffer
-	if err := json.Indent(&buf, []byte(bodyBytes), "", "    "); err != nil {
-		return "", fmt.Errorf("Failed to format the custom texts JSON: %w", err)
+	contentToEditJSON, err := json.MarshalIndent(contentToEdit, "", "    ")
+	if err != nil {
+		return "", fmt.Errorf("failed to serialize the prompt custom text to JSON: %w", err)
 	}
-	return buf.String(), nil
+
+	return string(contentToEditJSON), nil
 }
 
-func downloadBrandingTextLocale(filename string) map[string]interface{} {
-	url := fmt.Sprintf("%s/%s", textLocalesURL, filename)
-	resp, err := http.Get(url)
+// downloadDefaultBrandingTextTranslations will download all the prompt's possible
+// screen values. In case of encountering any errors it will simply ignore them
+// and let the user define by hand all the values for the screen.
+func downloadDefaultBrandingTextTranslations(prompt, language string) map[string]interface{} {
+	url := fmt.Sprintf(textLocalesURL, language)
+
+	response, err := http.Get(url)
 	if err != nil {
 		return nil
 	}
-	defer resp.Body.Close()
 
-	if resp.StatusCode == http.StatusOK {
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return nil
-		}
-		var result map[string]interface{}
-		if err := json.Unmarshal(body, &result); err != nil {
+	defer response.Body.Close()
+
+	if response.StatusCode == http.StatusOK {
+		var allPrompts []map[string]interface{}
+		if err := json.NewDecoder(response.Body).Decode(&allPrompts); err != nil {
 			return nil
 		}
-		return result
+
+		selectedPrompt := allPrompts[0][prompt].(map[string]interface{})
+
+		return selectedPrompt
 	}
+
 	return nil
 }
 
-func mergeBrandingTextLocales(d map[string]interface{}, c map[string]interface{}) map[string]map[string]interface{} {
-	result := make(map[string]map[string]interface{})
-
-	for p, pv := range d {
-		if translations, ok := pv.(map[string]interface{}); ok {
-			for k, v := range translations {
-				if !strings.HasPrefix(k, "error") && !strings.HasPrefix(k, "devKeys") {
-					if _, ok := result[p]; !ok {
-						result[p] = make(map[string]interface{})
-					}
-					if _, ok := result[p][k]; !ok {
-						result[p][k] = make(map[string]interface{})
-					}
-					result[p][k] = v
-				}
+func mergeBrandingTextTranslations(
```

Comment:
Just to double check, did you tests these changes manually?

Reply:
I just recorded the screen bit with the Mac's built-in functionality: Command + Shift + 5